transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/ws/ClosingHandshakeIT.java

```diff
@@ -0,0 +1,190 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaazing.gateway.transport.wsn.specification.ws;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+
+import java.net.URI;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+/**
+ * RFC-6455, section 7 "Closing the Connection"
+ */
+public class ClosingHandshakeIT {
+
+    private final K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing/specification/ws/closing");
+    
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/echo"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("http://localhost:8001")
+                            .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(8, SECONDS));
+
+    @Rule
+    public final TestRule chain = outerRule(k3po).around(gateway).around(timeout);
+
+    @Test
+    @Specification({
+        "client.send.empty.close.frame/handshake.request.and.frame"})
+    public void shouldCompleteCloseHandshakeWhenClientSendEmptyCloseFrame() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.frame.with.code.1000/handshake.request.and.frame"})
+    public void shouldCompleteCloseHandshakeWhenClientSendCloseFrameWithCode1000() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({
+        "client.send.close.frame.with.code.1000.and.reason/handshake.request.and.frame"})
+    public void shouldCompleteCloseHandshakeWhenClientSendCloseFrameWithCode1000AndReason() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("Read code different than expected and did not fail")
```

Comment:
@jitsni the expected is this
```
read [0x88 0x02 0x03 0xea]

closed
```
and actual is
```
read [0x88 0x16 0x03 0xe8]
```

The gateway is not closing at the script says it should. Where should I file an issue?